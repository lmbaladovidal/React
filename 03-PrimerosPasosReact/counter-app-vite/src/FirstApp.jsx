import PropTypes from 'prop-types'
const newMessage = 'Leandro';

function getObj(){
    return {Nombre:"Leandro",
            Apellido:"Balado Vidal"}
}

export const FirstApp = ({title, subtitle,name}) => {
  console.log(title)
  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        {/* <code>{JSON.stringify(getObj())}</code> */}
        <p>{subtitle}</p>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subtitle : PropTypes.string,
    counter : PropTypes.number
};

FirstApp.defaultProps = {
    name :"Leandro Martin",
    //title : "No Hay titulo",
    subtitle : "No Hay subtitulo"
}
