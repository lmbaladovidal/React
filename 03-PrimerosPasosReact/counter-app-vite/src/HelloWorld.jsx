export const HelloWorld = () => {
  return (
    <h2>HelloWorld</h2>
  )
}
