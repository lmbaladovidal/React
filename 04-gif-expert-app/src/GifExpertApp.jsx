import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Kimetsu no yaiba','My Hero Academy' ]);
  const onAddCategory = (newCategory)=>{
    setCategories([newCategory,...categories]);
  }
  return (
    <>
        {/* titulo */}
        <h1>GifExpertApp</h1>
        {/* input */}
        <AddCategory 
        /* onAddCategory={setCategories} */
        onNewCategory={onAddCategory}
        />
        {/* listado de gif */}
        <ol>          
          {categories.map((category,i)=>{
            return <li key={i}>{category}</li>
          })}
        </ol>
        {/* gif items */}
    </>
  )
}
