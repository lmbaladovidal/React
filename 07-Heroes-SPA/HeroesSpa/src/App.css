.btn-more {
    text-decoration: none;
    position: relative;
    font-size: 12px;
    text-transform: uppercase;
    text-decoration: none;
    padding: 1em 2.5em;
    display: inline-block;
    border-radius: 6em;
    transition: all .2s;
    border: none;
    font-family: inherit;
    font-weight: 500;
    color: black;
    background-color: white;
   }
   
   .btn-more:hover {
    text-decoration: none;
    transform: translateY(-3px);
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
   }
   
   .btn-more:active {
    transform: translateY(-1px);
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.2);
   }
   
   .btn-more::after {
    content: "";
    display: inline-block;
    height: 100%;
    width: 100%;
    border-radius: 100px;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -1;
    transition: all .4s;
   }
   
   .btn-more::after {
    background-color: #fff;
   }
   
   .btn-more:hover::after {
    transform: scaleX(1.4) scaleY(1.6);
    opacity: 0;
   }

   .card{
    margin-top: 10px;
    margin-left: 10px;
    padding-top: 5px;
    transition: 200ms;
    min-width: 300px;
    min-height: 300px;
   }
   .card:hover{
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
   }