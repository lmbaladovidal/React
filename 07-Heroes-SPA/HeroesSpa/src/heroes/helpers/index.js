export * from './getHeroesByPublisher';
export * from './getHeroesById';
export * from './getHeroByName';