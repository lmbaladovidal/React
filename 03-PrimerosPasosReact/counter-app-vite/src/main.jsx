import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { FirstApp } from './FirstApp';
import {HelloWorld} from './HelloWorld';
import {CounterApp} from './CounterApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorld />
        <CounterApp value={22}/>
        {/* <FirstApp title={"Hola, soy Goku!"} subtitle='Soy un sexy subtitulo' counter={1}/> */}
    </React.StrictMode>
);