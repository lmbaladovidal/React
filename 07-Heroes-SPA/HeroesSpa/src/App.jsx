import { useState } from 'react'
import './App.css'
import { AuthProvider } from './auth'
import { AppRouter } from './router/AppRouter'



export const App = ()=>{
  const [count, setCount] = useState(0)

    return (
      <AuthProvider>
        <AppRouter/>
      </AuthProvider>
  )
}

