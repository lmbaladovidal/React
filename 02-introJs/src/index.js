import heroes from './data/heroes'
console.log(heroes);

const getHeroesById = (id)=>{
    return heroes.find((heroe) => heroe.id === id);
}

const getHeroesByOwner = (owner)=>{
    return heroes.filter((heroe) => heroe.owner === owner);
}

console.log(getHeroesById(2));
console.log(getHeroesByOwner('DC'));
let arr = [0,1,2,3];
arr[10] = 10;
console.log(arr);
console.log(arr[9]);