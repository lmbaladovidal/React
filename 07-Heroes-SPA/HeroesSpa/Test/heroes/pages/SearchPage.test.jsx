import { fireEvent, getByRole, getByText, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { Search } from "../../../src/heroes";

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:()=>mockedUseNavigate
}))

describe('Pruebas sobre <NavBar/>', () => {
    
    beforeEach(()=> jest.clearAllMocks())

    test('should render correctly whit the default values', () => {

        const { container } = render(
            <MemoryRouter>
                <Search />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    })


    test('should render batman and the input with the queryString value', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Search />
            </MemoryRouter>
        );
        const input = screen.getByRole('textbox');
        const img = screen.getByRole('img');
        const searchHero = screen.getByLabelText('searchHero')
        expect(input.value).toBe('batman');
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
        expect(searchHero.style.display).toBe('none')
    })


    test('should show an error if the hero is not founded', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <Search />
            </MemoryRouter>
        );
        expect(screen.getByLabelText('noHero').style.display).toBe('');
        expect(screen.getByLabelText('searchHero').style.display).toBe('none');

    })

    test('should call the navigate to the new view', () => {
        const inputValue ='superman'
        render(
            <MemoryRouter initialEntries={['/search']}>
                <Search />
            </MemoryRouter>
        );
        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { name: 'searchText', value: inputValue } })
        const form = screen.getByLabelText('search-form');
        fireEvent.submit(form);
        expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`)


    })
})