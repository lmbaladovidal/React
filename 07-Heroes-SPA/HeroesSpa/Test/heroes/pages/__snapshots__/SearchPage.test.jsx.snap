// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <NavBar/> should render correctly whit the default values 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="search-form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="searchHero"
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        aria-label="noHero"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
